tool/release_ready.sh

```diff
@@ -0,0 +1,76 @@
+# Ensures that the package is ready for a release.
+# 
+# Will update the version.dart file and update the CHANGELOG.md.
+#
+# Run in the current directory with existing version
+# `./release_ready.sh`
+#
+# Try and run with new version
+# `./release_ready.sh <version>
+
+# Check if current directory has a pubspec.yaml, if so we assume it is correctly set up.
+if [ ! -f "pubspec.yaml" ]; then
+    echo "$(pwd) is not a valid package, missing pubspec.yaml."
+    exit 1
+fi
+
+currentBranch=$(git symbolic-ref --short -q HEAD)
+if [[ ! $currentBranch == "main" ]]; then
+    echo "Releasing is only supported on the main branch."
+    exit 1
+fi
+
+# Get package information
+package_version=$(dart pub deps --json | pcregrep -o1 -i '"version": "(.*?)"' | head -1)
+package_name=$(dart pub deps --json | pcregrep -o1 -i '"name": "(.*?)"' | head -1)
+
+# Get new version
+new_version="";
+if [ "$1" != "" ]; then
+  new_version="$1";
+else
+  new_version="$(dart pub deps --json | pcregrep -o1 -i '"version": "(.*?)"' | head -1)";
```

Comment:
Won't this cause line 35 to always be false?  If so we could fail fast here rather than on line 37